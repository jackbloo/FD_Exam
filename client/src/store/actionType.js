export const LOGIN = 'LOGIN'
export const FETCHPASSWORD = 'FETCHPASSWORD'
export const LOGOUT = 'LOGOUT'
export const FETCHDETAIL = 'FETCH DETAIL'
export const SETEDIT = 'SETEDIT'
export const SHOW = 'SHOW'
export const UNSHOW = 'UNSHOW'
export const LOAD = 'LOAD'
export const UNLOAD = 'UNLOAD'
export const DELETE = 'DELETE'