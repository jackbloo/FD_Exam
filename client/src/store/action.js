import {LOGIN, LOGOUT, FETCHPASSWORD,SHOW,UNSHOW,LOAD,UNLOAD, FETCHDETAIL} from '../store/actionType'


export const show = () => ({type:SHOW})
export const unshow = () => ({type:UNSHOW})
export const login = () => ({type:LOGIN})
export const logout = () => ({type:LOGOUT})
export const load = () => ({type:LOAD})
export const unload = () => ({type: UNLOAD})
export const fetchPassword = (data) =>  ({type:FETCHPASSWORD, data})


// export const deleteOnePass = (id) => dispatch => {
//     dispatch({type:LOAD})
//     pm
//     .doc(id)
//     .delete()
//     .then(function() {
//         dispatch({type:UNLOAD})
//         toast.success('Your Password is deleted')
//     }).catch(function(error) {
//         toast.error(error)
//     });
// }
