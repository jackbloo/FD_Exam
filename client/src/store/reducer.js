import {GETDATA} from './actionType'

const initialState = {
    editorChoice:[],
    latestArticles:[],
    latestReview:[],
    navbar: [{id:'skincare'},{id:'make up'},{id:'body'},{id:'hair'},{id:'fragrance'},{id:'nails'},{id:'tools'},{id:'brands'}],
    linkData:[{id: 'About Us'}, {id: 'Terms & Conditions'}, {id: 'Awards'},{id: 'Feedback'},{id: 'Privacy Policy'}, {id: 'Newsletter'}, {id: 'Content'},{id: 'Help'}]
}

const reducer = (state = initialState, action) => {
    switch(action.type){
        case GETDATA:
            const data = action.data
            return {
                ...state,
                editorChoice: [...data["editor's choice"]],
                latestArticles:[...data["latest articles"]],
                latestReview:[...data["latest review"]]
            }
        default:
            return state     
    }
 
}

export default reducer