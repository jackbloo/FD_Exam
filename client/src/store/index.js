import { configureStore } from '@reduxjs/toolkit'
import reducer from '../store/reducer'

const store = configureStore({reducer: reducer})

export default store