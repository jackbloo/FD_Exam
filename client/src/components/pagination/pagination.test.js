import React from 'react';
import {render, screen} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Pagination from './pagination';
import { Provider} from 'react-redux'
import store from '../../store'


describe('Unit Testing of Pagination', () => {
  
    it('Should Render Pagination match snapshot', ()=>{

     render(
            <Provider store={store}>
                    <Pagination/>
            </Provider>
        )
        const card = screen.getByTestId('pagination-container')
        expect(card).toMatchSnapshot()
    })


    
    it('Should Render Pagination without crashing', ()=>{
     render(
            <Provider store={store}>
                    <Pagination />
            </Provider>
        )
        const card = screen.getByTestId('pagination-container')
        expect(card).toBeInTheDocument()
        expect(card).toBeVisible()
    })

    
    it('Should Render pagination left arrow', ()=>{
     render(
            <Provider store={store}>
                    <Pagination />
            </Provider>
        )
        const card = screen.getByTestId('pagination-left-arrow')
        expect(card).toBeInTheDocument()
        expect(card).toBeVisible()
    })

    it('Should Render pagination right arrow', ()=>{
        render(
               <Provider store={store}>
                       <Pagination />
               </Provider>
           )
           const card = screen.getByTestId('pagination-right-arrow')
           expect(card).toBeInTheDocument()
           expect(card).toBeVisible()
       })

    it('Should Render pagination dots', ()=>{
        render(
               <Provider store={store}>
                       <Pagination />
               </Provider>
           )
           const card = screen.getByTestId('pagination-dots-0')
           expect(card).toBeInTheDocument()
           expect(card).toBeVisible()
       })

})

