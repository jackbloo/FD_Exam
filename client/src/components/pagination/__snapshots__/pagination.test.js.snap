// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Testing of Pagination Should Render Pagination match snapshot 1`] = `
<div
  class="sc-gsDKAQ hHEjbl"
  data-testid="pagination-container"
>
  <i
    class="sc-bdvvtL dAuLEb fa fa-arrow-left icon"
    data-testid="pagination-right-arrow"
  />
  <i
    class="sc-dkPtRN dhERQJ fa fa-circle icon"
    data-testid="pagination-dots-0"
  />
  <i
    class="sc-dkPtRN djnuNB fa fa-circle icon"
    data-testid="pagination-dots-1"
  />
  <i
    class="sc-dkPtRN djnuNB fa fa-circle icon"
    data-testid="pagination-dots-2"
  />
  <i
    class="sc-bdvvtL dAuLEb fa fa-arrow-right icon"
    data-testid="pagination-left-arrow"
  />
</div>
`;
